app/models/race.rb

```diff
@@ -51,7 +51,9 @@ def self.active
   end
 
   def self.friendly_find(slug)
-    where('LEFT(id::text, ?) = ?', slug.length, slug).order(created_at: :asc).first
+    race = where('LEFT(id::text, ?) = ?', slug.length, slug).order(created_at: :asc).first
+    raise ActiveRecord::RecordNotFound if race.nil?
```

Comment:
Since this method now raises on not finding, should it follow Rails find convention and be a bang method?  (ex: `find_by!`)

Reply:
Yeah I was torn here because bang -> raise is the overall Rails convention, but [`find`](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find) breaks that convention by raising, and I saw this method as closely related to `find`.

Probably bang is better though, so there's no question about it.

Reply:
I never really understood why all the `find_by` methods returned nil on not finding, but `find` would raise.  Seems like an easy thing to have be standardized.